Windows cmake build errors

@mrry I tried to build the HEAD and got the errors below:

```
"C:\src\tensorflow\tensorflow\contrib\cmake\build\tf_python_build_pip_package.vcxproj" (default target) (1) ->
"C:\src\tensorflow\tensorflow\contrib\cmake\build\pywrap_tensorflow.vcxproj" (default target) (3) ->
(ClCompile target) ->
  c:\src\tensorflow\tensorflow\contrib\cmake\build\pywrap_tensorflow.cc(3823): error C2440: 'return': cannot convert from 'tensorflow::Status' to 'bool' [C:\sr
c\tensorflow\tensorflow\contrib\cmake\build\pywrap_tensorflow.vcxproj]
  c:\src\tensorflow\tensorflow\contrib\cmake\build\pywrap_tensorflow.cc(3827): error C2440: 'return': cannot convert from 'tensorflow::Status' to 'bool' [C:\sr
c\tensorflow\tensorflow\contrib\cmake\build\pywrap_tensorflow.vcxproj]
  c:\src\tensorflow\tensorflow\contrib\cmake\build\pywrap_tensorflow.cc(5227): error C2660: 'TF_NewSession': function does not take 2 arguments [C:\src\tensorf
low\tensorflow\contrib\cmake\build\pywrap_tensorflow.vcxproj]
  c:\src\tensorflow\tensorflow\contrib\cmake\build\pywrap_tensorflow.cc(5343): error C2664: 'void TF_ExtendGraph(TF_DeprecatedSession *,const void *,size_t,TF_
Status *)': cannot convert argument 1 from 'TF_Session *' to 'TF_DeprecatedSession *' [C:\src\tensorflow\tensorflow\contrib\cmake\build\pywrap_tensorflow.vcxpr
oj]
  c:\src\tensorflow\tensorflow\contrib\cmake\build\pywrap_tensorflow.cc(5544): error C2664: 'void tensorflow::TF_Run_wrapper(TF_DeprecatedSession *,const TF_Bu
ffer *,PyObject *,const tensorflow::NameVector &,const tensorflow::NameVector &,TF_Status *,tensorflow::PyObjectVector *,TF_Buffer *)': cannot convert argument
 1 from 'TF_Session *' to 'TF_DeprecatedSession *' [C:\src\tensorflow\tensorflow\contrib\cmake\build\pywrap_tensorflow.vcxproj]
  c:\src\tensorflow\tensorflow\contrib\cmake\build\pywrap_tensorflow.cc(5709): error C2664: 'void tensorflow::TF_PRunSetup_wrapper(TF_DeprecatedSession *,const
 tensorflow::NameVector &,const tensorflow::NameVector &,const tensorflow::NameVector &,TF_Status *,const char **)': cannot convert argument 1 from 'TF_Session
 *' to 'TF_DeprecatedSession *' [C:\src\tensorflow\tensorflow\contrib\cmake\build\pywrap_tensorflow.vcxproj]
  c:\src\tensorflow\tensorflow\contrib\cmake\build\pywrap_tensorflow.cc(5804): error C2664: 'void tensorflow::TF_PRun_wrapper(TF_DeprecatedSession *,const char
 *,PyObject *,const tensorflow::NameVector &,TF_Status *,tensorflow::PyObjectVector *)': cannot convert argument 1 from 'TF_Session *' to 'TF_DeprecatedSession
 *' [C:\src\tensorflow\tensorflow\contrib\cmake\build\pywrap_tensorflow.vcxproj]
  c:\src\tensorflow\tensorflow\contrib\cmake\build\pywrap_tensorflow.cc(7501): error C2660: 'tensorflow::swig::RunCppShapeInference': function does not take 4
arguments [C:\src\tensorflow\tensorflow\contrib\cmake\build\pywrap_tensorflow.vcxproj]

    341 Warning(s)
    8 Error(s)
```

--- a/tensorflow/contrib/cmake/CMakeLists.txt
+++ b/tensorflow/contrib/cmake/CMakeLists.txt
@@ -28,9 +28,11 @@ option(tensorflow_BUILD_CONTRIB_KERNELS "Build OpKernels from tensorflow/contrib
 option(tensorflow_BUILD_CC_TESTS "Build cc unit tests " OFF)
 option(tensorflow_BUILD_PYTHON_TESTS "Build python unit tests " OFF)
 
-#Threads: defines CMAKE_THREAD_LIBS_INIT and adds -pthread compile option for
-# targets that link ${CMAKE_THREAD_LIBS_INIT}.
-find_package (Threads)
+if (NOT WIN32)
+  # Threads: defines CMAKE_THREAD_LIBS_INIT and adds -pthread compile option
+  # for targets that link ${CMAKE_THREAD_LIBS_INIT}.
+  find_package (Threads)
+endif()
 
 # [CLEANUP] Remove when done
 # For debugging

--- a/tensorflow/contrib/cmake/tf_python.cmake
+++ b/tensorflow/contrib/cmake/tf_python.cmake
@@ -473,10 +473,13 @@ add_dependencies(tf_python_ops tf_python_op_gen_main)
 
 find_package(SWIG REQUIRED)
 # Generate the C++ and Python source code for the SWIG wrapper.
+# NOTE(mrry): We always regenerate the SWIG wrapper, which means that we must
+# always re-link the Python extension, but we don't have to track the
+# individual headers on which the SWIG wrapper depends.
 add_custom_command(
       OUTPUT "${CMAKE_CURRENT_BINARY_DIR}/tf_python/tensorflow/python/pywrap_tensorflow.py"
              "${CMAKE_CURRENT_BINARY_DIR}/pywrap_tensorflow.cc"
-      DEPENDS tf_python_touchup_modules
+      DEPENDS tf_python_touchup_modules __force_rebuild
       COMMAND ${SWIG_EXECUTABLE}
       ARGS -python -c++
            -I${tensorflow_source_dir}

